### PR TITLE
tests: adding support for debian 10 on gce

### PR DESCRIPTION
**Pull request by sergiocazzolato**
New image created and support added to snapd test suite to test on debian 10

### CODE REVIEW

File: tests/main/system-usernames/task.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ environment:
     # List of expected snap install failures due to libseccomp/golang-seccomp
     # being too old. This should only reduce with time since new systems should
     # have newer libseccomp and golang-seccomp
-    EXFAIL: "centos-7-64 debian-9-64 fedora-29-64 fedora-30-64 opensuse-15\\.0-64 ubuntu-14"
+    EXFAIL: "centos-7-64 debian-9-64 debian-10-64 fedora-29-64 fedora-30-64 opensuse-15\\.0-64 ubuntu-14"
```
Comment (bboozzoo) [inline tests/main/system-usernames/task.yaml:11]: Hmmm this is rather unexpected. I think debian 10 should carry libseccomp/golang-seccomp versions that are much more recent than ones in 9.

File: tests/main/system-usernames-illegal/task.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ summary: ensure unapproved user cannot be used with system-usernames
 # List of expected snap install failures due to libseccomp/golang-seccomp being
 # too old. Since the illegal name check happens after verifying system support,
 # we can ignore these.
-systems: [-amazon-linux-2-*, -centos-7-*, -debian-9-*, -fedora-29-*, -fedora-30-*, -opensuse-15.2-*, -opensuse-15.1-*, -ubuntu-14.04-*]
+systems: [-amazon-linux-2-*, -centos-7-*, -debian-9-*, -debian-10-*, -fedora-29-*, -fedora-30-*, -opensuse-15.2-*, -opensuse-15.1-*, -ubuntu-14.04-*]
```
Comment (bboozzoo) [inline tests/main/system-usernames-illegal/task.yaml:6]: It is unexpected to see debian 10 here, we need to investigate.

Fedora 32 is the oldest release we have in spread.yaml. I think we can garbage collect fedora-(29|30) here.

File: tests/main/system-usernames-missing-user/task.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ summary: ensure snap fails to install if one of user or group doesn't exist
 # too old. Since the illegal name check happens after verifying system support,
 # we can ignore these. Ignore ubuntu-core since groupdel doesn't support
 # --extrausers
-systems: [-amazon-linux-2-*, -centos-7-*, -debian-9-*, -fedora-29-*, -fedora-30-*, -opensuse-15.2-*, -opensuse-15.1-*, -ubuntu-14.04-*, -ubuntu-core-*]
+systems: [-amazon-linux-2-*, -centos-7-*, -debian-9-*, -debian-10-*, -fedora-29-*, -fedora-30-*, -opensuse-15.2-*, -opensuse-15.1-*, -ubuntu-14.04-*, -ubuntu-core-*]
```
Comment (bboozzoo) [inline tests/main/system-usernames-missing-user/task.yaml:7]: Same here about old Fedora releases.

File: tests/main/system-usernames/task.yaml (hunk 1 above)
Comment (bboozzoo) [inline tests/main/system-usernames/task.yaml:11]: Old fedora releases can be dropped here too

**Review by bboozzoo — COMMENTED** [review]
Some comments. Pushed a patch to enable debian-10-64 in github workflow, and switched debian-9 to manual in spread.yaml

**Review by mvo5 — APPROVED** [review]
Thank you! Looks fine.

File: tests/main/system-usernames-missing-user/task.yaml (hunk 3 above)
Comment (bboozzoo) [reply tests/main/system-usernames-missing-user/task.yaml]: This was addressed in #10257

**Comment by bboozzoo** [comment]
```
google:debian-10-64 .../tests/main/interfaces-timeserver-control# systemctl status systemd-timesyncd
● systemd-timesyncd.service - Network Time Synchronization
   Loaded: loaded (/lib/systemd/system/systemd-timesyncd.service; enabled; vendor preset: enabled)
  Drop-In: /usr/lib/systemd/system/systemd-timesyncd.service.d
           └─disable-with-time-daemon.conf
   Active: inactive (dead)
Condition: start condition failed at Mon 2021-05-10 11:07:40 UTC; 2min 32s ago
           └─ ConditionFileIsExecutable=!/usr/sbin/chronyd was not met
     Docs: man:systemd-timesyncd.service(8)

May 10 10:52:08 may101042-091747 systemd[1]: Condition check resulted in Network Time Synchronization being skipped.
May 10 11:05:36 may101042-091747 systemd[1]: Condition check resulted in Network Time Synchronization being skipped.

```

Looks like we need to purge chrony from the image first.

**Comment by bboozzoo** [comment]
> Now I see the error:
> 
> -- The job identifier is 728.
> May 11 22:20:50 may112217-126969 systemd-modules-load[1253]: Failed to find module 'lp'
> May 11 22:20:50 may112217-126969 systemd-modules-load[1253]: Failed to find module 'ppdev'
> May 11 22:20:50 may112217-126969 systemd-modules-load[1253]: Failed to find module 'parport_pc'
> May 11 22:20:50 may112217-126969 systemd[1]: systemd-modules-load.service: Main process exited, code=exited, status=1/FAILURE

Hmm those should be part of the linux-image package AFAIU?

**Comment by bboozzoo** [comment]
Soo, the certain modules are apparently missing from the cloud kernel which we have. The exact version is `4.19.0-16-cloud-amd64`. Looks like `lp`, `parport_pc` and `ppdev` are among those, which kind of makes sense. I mean, nobody would expect a parallel port, or a local printer drivers to be of any use in a cloud image which is running in some random VM in a data center.
I think we can just reset the failed status of systemd-modules-load service. I'll push a patch

**Review by bboozzoo — APPROVED** [review]
Should be good to land now.

**Pull request merged**